value/src/main/java/com/google/auto/value/processor/AutoValueProcessor.java

```diff
@@ -216,6 +216,7 @@ private String generatedSubclassName(TypeElement type, int depth) {
 
     private ImmutableList<String> buildAnnotations(TypeSimplifier typeSimplifier) {
       ImmutableList.Builder<String> builder = ImmutableList.builder();
+      Set<Name> methodAnnotations = null;
```

Comment:
I just tried it, it doesn't work at all, at least for javac. The same (source-code) annotation as method annotionMirror is not even equal() to the annotationMirror returned for the type.